### PR TITLE
feat: IPFS plugin and app

### PR DESCRIPTION
**Pull request by ryanio**
#### What does it do?
* Adds `IPFS` plugin and app
* Adds a `beforeStart` hook for plugins that can `execute` commands. Used in this case for setting HTTP CORS config for the daemon.

Steps to test:
1. Turn on IPFS
2. Go to `APPS` tab and open `IPFS Web UI`.
3. Check to make sure it's connected. You can try uploading a file as well on the `Files` tab.

image

image

--

Should be merged with https://github.com/ethereum/grid-ui/pull/102

### CODE REVIEW

File: ethereum_clients/Plugin.js [hunk 1]
```diff
@@ -180,7 +180,14 @@ class Plugin extends EventEmitter {
 
   async start(release, flags, config) {
     // TODO do flag validation here based on proxy metadata
-
+    if (this.config.beforeStart) {
+      if (this.config.beforeStart.execute) {
```
Comment (marcgarreau) [inline ethereum_clients/Plugin.js]: nit: shed one layer by combining: `if (this.config.beforeStart && this.config.beforeStart.execute)`

Comment (ryanio) [reply ethereum_clients/Plugin.js]: thx

**Comment by ryanio** [comment]
thanks, i handled the blocking error dialog issue with https://github.com/ethereum/grid/pull/321/commits/e5c38d22656876265f6e0e32fc56930c72787ed8 

I also get `> 17:29:01.113 ERROR core: failure on stop: context canceled builder.go:47` consistently on closing down the process, but I figure it's non blocking for now. It looks like `stop()` in `ControlledProcess.js` calls `this.proc.kill('SIGINT')` so it shouldn't be closing it forcibly by any means.

**Comment by evertonfraga** [comment]
the plugin is not filtering by OS.

i'll send a PR

image

**Comment by ryanio** [comment]
watchForPeers bug fixed in https://github.com/ethereum/grid-ui/pull/89/commits/a7332ec0623fde2808da7a49b9264ec7f073bcfc

**Pull request merged**